gateway/src/main/java/io/zeebe/gateway/impl/job/LongPollingActivateJobsHandler.java

```diff
@@ -15,29 +17,38 @@
 import io.zeebe.gateway.protocol.GatewayOuterClass.ActivateJobsResponse;
 import io.zeebe.util.sched.Actor;
 import io.zeebe.util.sched.ScheduledTimer;
+import io.zeebe.util.sched.clock.ActorClock;
 import java.time.Duration;
 import java.util.HashMap;
-import java.util.LinkedList;
 import java.util.Map;
+import java.util.Objects;
 import java.util.Queue;
 import org.slf4j.Logger;
 
 public class LongPollingActivateJobsHandler extends Actor {
 
-  public static final Duration DEFAULT_LONG_POLLING_TIMEOUT = Duration.ofSeconds(10);
   private static final String JOBS_AVAILABLE_TOPIC = "jobsAvailable";
-  private static final int NO_JOBS_RESPONSE_THRESHOLD = 3;
   private static final Logger LOG = Loggers.GATEWAY_LOGGER;
 
   private final ActivateJobsHandler activateJobsHandler;
   private final BrokerClient brokerClient;
-  private final Map<String, Integer> zeroResponses = new HashMap<>();
-  private final Map<String, Queue<LongPollingActivateJobsRequest>> blockedRequests =
-      new HashMap<>();
 
-  public LongPollingActivateJobsHandler(BrokerClient brokerClient) {
+  // jobType -> state
+  private final Map<String, JobTypeAvailabilityState> jobTypeState = new HashMap<>();
+  private final Duration longPollingTimeout;
+  private final long probeTimeoutMillis;
```

Comment:
```suggestion
  private final Duration probeTimeout;
```

Reply:
it is better be `long` because it is used for comparing the last updated time in `probe()`.